paddle/fluid/operators/collective/recv_v2_op.cu.cc

```diff
@@ -99,7 +104,66 @@ class RecvOpV2CUDAKernel : public framework::OpKernel<T> {
     auto out_dims = out->dims();
     auto numel = out->numel();
 
-    out->mutable_data<T>(out_dims, place);
+    if (dynamic_shape) {
+      VLOG(3) << "recv_v2 will use dynamic shape with send_v2";
```

Comment:
It's too long.Wrap it as a function?

Reply:
is there any other place we will reuse this function with all these gpu<->cpu copies